charts/spire/charts/spire-server/values.yaml

```
@@ -73,12 +73,22 @@ affinity: {}
 
 topologySpreadConstraints: []
 
-dataStorage:
+# Used by sqlite3 for spire-server, by the disk KeyStore, and/or by Tornjak.
+persistence:
   enabled: true
   size: 1Gi
   accessMode: ReadWriteOnce
   storageClass: null
 
+dataStore:
+  sql:
+    # -- Set any setting from https://github.com/spiffe/spire/blob/v1.6.3/doc/plugin_server_datastore_sql.md under this section.
+    plugin_data:
+      # -- Set to either "postgres" or "mysql" if you want to use an external databse
+      database_type: ""
+      # -- Set to the appropriate setting for your chosen external database. Passwords will be protected
+      connection_string: ""
```

For documentation purposes and less hidden magic it might be better to default this to sqlite.

```suggestion
dataStore:
  sql:
    # -- Set any setting from https://github.com/spiffe/spire/blob/v1.6.3/doc/plugin_server_datastore_sql.md under this section.
    plugin_data:
      # -- Other supported databases are "postgres" and "mysql"
      database_type: sqlite3
      # -- The connectionstring to connect to your database instance.
      connection_string: /run/spire/data/datastore.sqlite3
```

---

The logic here was developed with the idea that internal databases as per https://github.com/spiffe/helm-charts/pull/233 could be a thing. The suggested settings somewhat conflict with that.